Libraries/LibJS/Runtime/ArrayPrototype.cpp

```diff
@@ -447,6 +448,65 @@ Value ArrayPrototype::reduce(Interpreter& interpreter)
     return accumulator;
 }
 
+Value ArrayPrototype::reduceRight(Interpreter& interpreter)
+{
+    auto* this_object = interpreter.this_value().to_object(interpreter);
+    if (!this_object)
+        return {};
+
+    auto initial_length = get_length(interpreter, *this_object);
+    if (interpreter.exception())
+        return {};
+
+    int start = initial_length - 1;
+
+    auto accumulator = js_undefined();
+    if (interpreter.argument_count() > 1) {
+        accumulator = interpreter.argument(1);
+    } else {
+        bool start_found = false;
+        while (!start_found && start >= 0) {
+            auto value = this_object->get_by_index(start);
+            if (interpreter.exception())
+                return {};
+            start_found = !value.is_empty();
+            if (start_found)
+                accumulator = value;
+            start -= 1;
+        }
+        if (!start_found) {
+            interpreter.throw_exception<TypeError>("Reduce of empty array with no initial value");
+            return {};
+        }
+    }
+
+    auto* callback_function = callback_from_args(interpreter, "reduceRight");
+    if (!callback_function)
+        return {};
```

Comment:
Can we move this (and probably also the one in `reduce` after the `get_length()` call but before the if/else? Not that it *really* matters as both are type errors, but that way the "missing/invalid callback" error is thrown before the "Reduce of empty array with no initial value" one, which is closer to the spec and other implementations :)